### DISCUSSION TITLE
Support Debian-based images

### DISCUSSION THREAD

**Discussion by bap14** [wardenenv/warden#578] [Polls]
While I appreciate the initiative to keep image sizes small, there are some compatibility issues with projects because of Alpine linux. Specifically Alpine-based images have similar but not necessarily the same version or implementation of common tools (e.g. gzip).

Some projects, namely Adobe Commerce, with their ECE Tools package assumes its being run in a Debian environment. This can make using those tools locally difficult if not impossible.

Given that PHP supports both Debian and Alpine based images, and we are (or will be) utilizing GitHub actions to build, is there a downside to building both flavors?

**Comment by navarr** [comment]
It might behoove us to give up on Alpine and switch to whatever the base PHP images run in general.

**Reply by bap14** [reply]
Well that's the thing. The native PHP images support both Debian (bullseye, buster, etc.) and Alpine. So we could do both. It would just mean moving some of the setup scripting from the Dockerfile itself into a build script (similar to the script to install PHP extensions).
